folder/folderutil.go

```diff
@@ -148,3 +150,19 @@ func HomeDirOrDefault(defaultDirectory string) string {
 	}
 	return usr.HomeDir
 }
+
+func UserConfigDirOrDefault(defaultConfigDir string) string {
+	userConfigDir, err := os.UserConfigDir()
+	if err != nil {
+		return defaultConfigDir
+	}
+	return userConfigDir
+}
+
+func UserAppConfigDirOrDefault(defaultAppConfigDir string) string {
+	userConfigDir := UserConfigDirOrDefault("")
+	if userConfigDir == "" {
+		return defaultAppConfigDir
+	}
+	return filepath.Join(userConfigDir, fileutil.ExecutableName())
+}
```

Comment:
@dogancanbakir while `fileutil.ExecutableName()` is good this will cause issues with 
library usage ex: uncover library imported in nuclei etc or if tool is renamed etc
i think it would be best if we just accept toolname as function parameter
```go
func UserAppConfigDirOrDefault(defaultAppConfigDir ,toolName string) string {
    ...
}
```